plugins/incremental-ingestion-backend/README.md

```diff
@@ -0,0 +1,328 @@
+
+Unfortunately, these two kinds of mutations are insufficient for very large data sources for the following reasons,
+
+1. Even when the API provides DELETE events, we still need a way to create the initial list of entities. For example, if you ingest all repositories from GitHub into Backstage and you use webhooks, you still need the initial list of entities.
+2. A `delta` mutation can not guarantee that mutations will not be missed. For example, if your Backstage portal is down while receiving a DELETE event, you might miss the event which leaves your catalog in an unclear state. How can you replay the missed events? Some data sources, like GitHub, provide an API for replaying missed events, but this increases complexity and is not available on all APIs.
+3. Addressing the above two use case with `full` mutation is not an option on very large datasets because a `full` mutation requires that all entities are in memory to create a diff. If your data source has 100k+ records, this can easily cause your processes to run out of memory.
+4. In cases when you can use `full` mutation, committing many entities into the processing pipeline fills up the processing queue and delays the processing of entities from other entity providers.
+
+We created the Incremental Entity Provider to address all of the above issues. The Incremental Entity Provider addresses these issues with a combination of `delta` mutations and a mark-and-sweep mechanism. Instead of doing a single `full` mutation, it performs a series of bursts. At the end of each burst, the Incremental Entity Provider performs the following three operations,
+
+1. Marks each received entity in the database.
+2. Annotates each entity with `backstage/incremental-entity-provider: <entity-provider-id>` annotation.
+3. Commits all of the entities with a `delta` mutation.
+
+Incremental Entity Providers will wait a configurable interval before proceeding to the next burst.
+
+Once the source has no more results, Incremental Entity Provider compares all entities annotated with `frontside/incremental-entity-provider: <entity-provider-id>` against all marked entities to determine which entities committed by same entity provider were not marked during the last ingestion cycle. All unmarked entities are deleted at the end of the cycle. The Incremental Entity Provider rests for a fixed internal before restarting the ingestion process.
```

Comment:
`frontside/incremental-entity-provider: <entity-provider-id>` needs to be changed to `backstage/incremental-entity-provider: <entity-provider-id>`

Reply:
Done!